js/components/frame.js

```diff
@@ -819,7 +820,9 @@ class Frame extends ImmutableComponent {
     })
     this.webview.addEventListener('page-favicon-updated', (e) => {
       if (e.favicons && e.favicons.length > 0) {
-        windowActions.setFavicon(this.frame, e.favicons[0])
+        imageUtil.getWorkingImageUrl(e.favicons[0], (imageFound) => {
```

Comment:
This is a pretty elegant way to do it, which avoids the complexity of promises. Good job :smile: